src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/MobCondenser.java

```diff
@@ -0,0 +1,111 @@
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHack;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHackConfig;
+import org.bukkit.Bukkit;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.entity.CreatureSpawnEvent;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemStack;
+import vg.civcraft.mc.civmodcore.entities.EntityUtils;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Random;
+
+public class MobCondenser extends BasicHack
+{
+	private Random rng;
+	private Map<EntityType, Double> mobSpawnModifiers;
+
+	public MobCondenser(SimpleAdminHacks plugin, BasicHackConfig config)
+	{
+		super(plugin, config);
+		this.rng = new Random();
+		this.mobSpawnModifiers = new HashMap<>();
+	}
+
+	@Override
+	public void onEnable()
+	{
+		super.onEnable();
+
+		final ConfigurationSection base = this.config.getBase();
+		final ConfigurationSection spawnModifiers = base.getConfigurationSection("mobSpawnModifiers");
+		if (spawnModifiers != null) {
+			for (final String key : spawnModifiers.getKeys(false)) {
+				final EntityType type = EntityUtils.getEntityType(key);
+				if (type == null) {
+					this.plugin.warning("[" + getClass().getSimpleName() + "] EntityType: \"" + key + "\" does not exist, skipping.");
+					continue;
+				}
+				double modifier = spawnModifiers.getDouble(key, 1.0d);
+				if (modifier < 0.0d) {
+					this.plugin.warning("[" + getClass().getSimpleName() + "] Mob Spawn Modifier: \"" + modifier + "\" for \"" + key + "\" is unsupported, defaulting to 1.0");
+					modifier = 1.0d;
+				}
+				this.mobSpawnModifiers.put(type, modifier);
+				this.plugin.info("[" + getClass().getSimpleName() + "] Registered Mob Spawn Modifier: [" + type + ": " + modifier + "]");
+			}
+		}
+	}
+
+	@Override
+	public void onDisable()
+	{
+		this.mobSpawnModifiers.clear();
+
+		super.onDisable();
+	}
+
+	private boolean roll(double chance) {
+		return rng.nextDouble() <= chance;
+	}
+
+	@EventHandler(priority = EventPriority.LOW) //Run before the PortalSpawnModifier
+	public void onCreatureSpawn(CreatureSpawnEvent e) {
+		if (mobSpawnModifiers.containsKey(e.getEntityType())) {
+			if (!roll(mobSpawnModifiers.get(e.getEntityType()))) {
+				if (e.getSpawnReason() != CreatureSpawnEvent.SpawnReason.SPAWNER_EGG) {
+					e.setCancelled(true);
+				}
+			}
+		}
+	}
+
+	@EventHandler
+	public void onMobEggUse(PlayerInteractEvent e) {
+		if (e.getAction() == Action.RIGHT_CLICK_BLOCK && e.getItem() != null) {
+			try {
+				EntityType type = EntityType.valueOf(e.getItem().getType().toString().replace("_SPAWN_EGG", ""));
```

Comment:
You can do: 
```java
EntityType type = SpawnEggUtils.getEntityType(e.getItem().getType());
```

Reply:
Thanks, I must've not seen that when I looked to see if there was something that does that.